packages/opal-server/opal_server/worker.py

```diff
@@ -46,11 +92,23 @@ def create_worker() -> Worker:
     worker = Worker(
         base_dir=opal_base_dir,
         scopes=ScopeRepository(RedisDB(opal_server_config.REDIS_URL)),
+        pubsub_client=PubSubClient(
+            server_uri=opal_client_config.SERVER_PUBSUB_URL,
+            extra_headers=[get_authorization_header(opal_server_config.OPAL_WS_TOKEN)],
+        ),
     )
 
     return worker
 
 
+def with_worker(f):
+    async def _inner(*args, **kwargs):
+        async with create_worker() as worker:
+            await f(worker, *args, **kwargs)
+
+    return _inner
```

Comment:
you can probably revert this as well (you don't need the async context manager anymore